plugins/multiexternalwrench/src/applymultiexternalwrench.cpp

```diff
@@ -0,0 +1,175 @@
+            m_rpcThread.setRobotModel(_model);
+            configuration_loaded = true;
+        } else {
+            yError ( "ERROR trying to get robot configuration file" );
+            return;
+        }
+    } else {
+            this->robotName = _model->GetName();
+            m_rpcThread.setRobotName ( robotName );
+            configuration_loaded = true;
+    }
+
+    // Starting RPC thread to read desired wrench to be applied
+    if ( !m_rpcThread.start() ) {
+        yError ( "ERROR: rpcThread did not start correctly" );
+    }
+    
+    this->m_updateConnection = event::Events::ConnectWorldUpdateBegin(boost::bind(&ApplyMultiExternalWrench::applyWrenchs,this));
+}
+
+}
+
+void ApplyMultiExternalWrench::applyWrenchs()
+{
+    //Now check for duration done flag from all the wrench threads and removes the ones that are done
+    //yInfo() << "Applying external wrenches";
+    //yInfo() << "Number of external wrenches : " << m_rpcThread.wrenchThreads.size();
+    m_rpcThread.m_lock.lock();
+    for(int i = 0; i < m_rpcThread.wrenchesVectorPtr->size() ; i++)
+    {
+        bool duration_check = m_rpcThread.wrenchesVectorPtr->at(i)->duration_done;
+        if(duration_check==false)
+        {
+            m_rpcThread.wrenchesVectorPtr->at(i)->applyWrench();
+        }
+        else
+        {
+            //yInfo() << "External wrench duration done";
+        }
+    }
+    m_rpcThread.m_lock.unlock();
+}
+
+void RPCServerThread::setRobotModel(physics::ModelPtr robotModel)
+{
+    m_robotModel = robotModel;
+}
+
+
+bool RPCServerThread::threadInit()
+{
+
+    if ( !m_rpcPort.open ( std::string ( "/"+m_robotName + "/applyMultiExternalWrench/rpc:i" ).c_str() ) ) {
+        yError ( "ERROR opening RPC port /applyExternalWrench" );
+        return false;
+    }
+  
+    return true;
+}
+
+void RPCServerThread::run()
+{
+    while(!isStopping())
+    {
+        yarp::os::Bottle command;
+        m_rpcPort.read ( command,true );
+        if(command.get(0).asString() == "help")
+        {
+            this->m_reply.addVocab ( yarp::os::Vocab::encode ( "many" ) );
+            this->m_reply.addString ( "Insert a command with the following format:" );
+            this->m_reply.addString ( "[link] [force] [torque] [duration]" );
+            this->m_reply.addString ( "e.g. chest 10 0 0 0 0 0 1");
+            this->m_reply.addString ( "[link]:     (string) Link ID of the robot as specified in robot's SDF" );
+            this->m_reply.addString ( "[force]:    (double x, y, z) Force components in N w.r.t. world reference frame" );
+            this->m_reply.addString ( "[torque]:   (double x, y, z) Torque components in N.m w.r.t world reference frame" );
+            this->m_reply.addString ( "[duration]: (double) Duration of the applied force in seconds" );
+            this->m_reply.addString ( "Note: The reference frame is the base/root robot frame with x pointing backwards and z upwards.");
```

Comment:
@traversaro I agree with you, the reference is gazebo world frame. Also a better additional note, in my opinion, is to indicate that external wrench is applied at the center of mass of the link(not at the link frame).